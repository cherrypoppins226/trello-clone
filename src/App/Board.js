import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Modal from "@material-ui/core/Modal";
import PropTypes from "prop-types";
import Popover from "@material-ui/core/Popover";
import CardsList from "./CardsList";
import QuickEditCard from "./QuickEditCard";
import ActionsMenu from "./CardsList/ActionsMenu";
import EditCard from "./EditCard";
import { cardDescription } from "./CardsList/Card";

const styles = {
  root: {
    padding: 8,
    margin: 0,
    height: "100%",
    width: "100%"
  },
  cardListContainer: {
    width: 280
  },
  cardsListActionsPopover: {
    backgroundColor: "transparent"
  }
};

class Board extends React.Component {
  state = {
    listBeingEdited: null,
    cardBeingQuickEdited: null,
    cardBeingEdited: null
  };

  render() {
    const { classes, lists } = this.props;
    const cardTitle = (v => (v ? cardDescription(v) : ""))(
      this.state.cardBeingEdited || this.state.cardBeingQuickEdited
    );
    return (
      <>
        <Grid container spacing={16} wrap="nowrap" className={classes.root}>
          {Object.entries(lists).map(([k, v]) => {
            return (
              <Grid item key={k} className={classes.cardListContainer}>
                <CardsList
                  title={k}
                  cards={v}
                  onQuickEditCard={card =>
                    this.setState({ cardBeingQuickEdited: card })
                  }
                  onEditCard={card => this.setState({ cardBeingEdited: card })}
                  onEditList={list => this.setState({ listBeingEdited: list })}
                />
              </Grid>
            );
          })}
        </Grid>
        <Popover
          anchorEl={this.state.listBeingEdited}
          container={this}
          open={Boolean(this.state.listBeingEdited)}
          onClose={_ => this.setState({ listBeingEdited: null })}
          TransitionProps={{ timeout: 0 }}
          elevation={1}
          anchorOrigin={{ vertical: "bottom", horizontal: "left" }}
          transformOrigin={{ vertical: "top", horizontal: "left" }}
          PaperProps={{ classes: { root: classes.cardsListActionsPopover } }}
        >
          <ActionsMenu
            onMenuItemClick={_ => this.setState({ listBeingEdited: null })}
          />
        </Popover>
        <Modal
          container={this}
          open={Boolean(this.state.cardBeingQuickEdited)}
          onClose={_ => this.setState({ cardBeingQuickEdited: null })}
          BackdropProps={{ id: "quickEditCardBackdrop" }}
        >
          <QuickEditCard
            title={cardTitle}
            anchorEl={this.state.cardBeingQuickEdited}
          />
        </Modal>
        <Modal
          container={this}
          open={Boolean(this.state.cardBeingEdited)}
          onClose={_ => this.setState({ cardBeingEdited: null })}
          BackdropProps={{ id: "editCardBackdrop" }}
        >
          <EditCard title={cardTitle} />
        </Modal>
      </>
    );
  }
}

const View = withStyles(styles)(Board);

View.propTypes = {
  lists: PropTypes.object.isRequired
};

export default View;
