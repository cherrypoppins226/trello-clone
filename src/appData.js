import faker from "faker";

faker.seed(1);

export const testData = {
  [faker.lorem.sentence()]: [faker.lorem.sentence(), faker.lorem.sentence()],
  [faker.lorem.sentence()]: [faker.lorem.sentence(), faker.lorem.sentence()]
};

export const appData = {};
for (let i = 0; i < 6; i++) {
  const key = faker.lorem.sentence();
  appData[key] = [];
  for (let j = 0; j < faker.random.number({ min: 2, max: 50 }); j++) {
    appData[key].push(faker.lorem.sentence());
  }
}
