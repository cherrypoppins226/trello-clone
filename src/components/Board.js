import React from "react";
import path from "path";
import { withStyles } from "material-ui/styles";
import Grid from "material-ui/Grid";
import PropTypes from "prop-types";
import CardsList from "./CardsList";
import EditCard from "./EditCard";

const styles = {
  grid: {
    padding: 8,
    margin: 0,
    height: "100%",
    width: "100%"
  },
  cardListContainer: {
    width: 280
  }
};

const View = class extends React.Component {
  state = { cardBeingEdited: null };

  render() {
    const { classes, lists } = this.props;
    return (
      <>
        <Grid container spacing={16} wrap="nowrap" className={classes.grid}>
          {Object.entries(lists).map(([k, v]) => {
            return (
              <Grid item key={k} className={classes.cardListContainer}>
                <CardsList
                  title={k}
                  cards={v}
                  onEditCard={card => this.setState({ cardBeingEdited: card })}
                />
              </Grid>
            );
          })}
        </Grid>
        <EditCard
          card={this.state.cardBeingEdited}
          onClose={_ => this.setState({ cardBeingEdited: null })}
        />
      </>
    );
  }
};

View.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.object.isRequired,
  lists: PropTypes.object.isRequired
};

View.displayName = path.basename(__filename, path.extname(__filename));

export default withStyles(styles)(View);
