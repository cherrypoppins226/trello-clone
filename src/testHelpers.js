import { render as renderer } from "react-testing-library";
import { createContext } from "react-cosmos-loader";

export const render = (fixture, proxies = []) => {
  const { mount, ...context } = createContext({ renderer, fixture, proxies });

  return (async () => {
    await mount();
    return { ...context.getWrapper(), ...context, rerender: mount };
  })();
};

export const NativeEvents = {
  mouse: {
    click: new MouseEvent("click", { bubbles: true, cancelable: true })
  }
};

export const getByRole = (node, role) => {
  return node.querySelector(`[role="${role}"]`);
};

export const getByAriaLabelled = (node, id) => {
  return node.querySelector(`[aria-labelledby="${id}"]`);
};

export const getByAriaDescribed = (node, id) => {
  return node.querySelector(`[aria-describedby="${id}"]`);
};
