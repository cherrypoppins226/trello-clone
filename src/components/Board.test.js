import React from "react";
import { render } from "../testHelpers.js";
import { testData } from "../appData.js";
import Board from "./Board";

jest.mock("./CardsList", () => () => "CardsList");
jest.mock("./EditCard", () => () => "EditCard");

it("initial render", () => {
  const { container } = render(<Board lists={testData} />);
  expect(container).toMatchSnapshot();
});
