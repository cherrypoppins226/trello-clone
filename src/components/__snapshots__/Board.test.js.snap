// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial render 1`] = `
<div>
  <div
    class="MuiGrid-container MuiGrid-spacing-xs-16 MuiGrid-wrap-xs-nowrap Board-grid-1"
  >
    <div
      class="MuiGrid-item Board-cardListContainer-2"
    >
      CardsList
    </div>
    <div
      class="MuiGrid-item Board-cardListContainer-2"
    >
      CardsList
    </div>
  </div>
  EditCard
</div>
`;
